projects/RabbitMQ.Client/client/api/AmqpTcpEndpoint.cs

```diff
@@ -76,7 +76,7 @@ public AmqpTcpEndpoint(string hostName, int portOrMinusOne, SslOption ssl, uint
             HostName = hostName;
             _port = portOrMinusOne;
             Ssl = ssl;
-            _maxMessageSize = maxMessageSize;
+            _maxMessageSize = Math.Min(maxMessageSize, ConnectionFactory.DefaultMaxMessageSize);
```

Comment:
Related to the comment below. 
This change is not necessary - If we continue to use 0 as the default value, i.e. unlimited, based on the comment.